src/tables.jl

```diff
@@ -0,0 +1,79 @@
+const AG = ArchGDAL
+
+struct Table{T} 
+    layer::T
+end
+
+function Table(layer::T) where {T<:Union{IFeatureLayer, FeatureLayer}}
+    featuredefn = layerdefn(layer)
+    ngeometries = ngeom(featuredefn)
+    if ngeometries == 0
+        print("NULL Geometry found")
+    end
+    Table{T}(layer)
+end
+
+function Tables.schema(layer::AG.AbstractFeatureLayer)
+    nfield = AG.nfield(layer)
+    featuredefn = AG.layerdefn(layer)
+    fielddefns = (AG.getfielddefn(featuredefn, i) for i in 0:nfield-1)
+    names_fields = Tuple(AG.getname(fielddefn) for fielddefn in fielddefns)
+    types_fields = Tuple(AG._FIELDTYPE[AG.gettype(fielddefn)] for fielddefn in fielddefns)
+    Tables.Schema(names_fields, types_fields)
+end
+
+function Base.iterate(t::Table, st = 0)
+    layer = t.layer
+    if iszero(st) 
+        AG.resetreading!(layer) 
+    end
+
+    nfeat = AG.nfeature(layer)
+    nfield = AG.nfield(layer)
+    ngeom = AG.ngeom(layer)
+    featuredefn = AG.layerdefn(layer)
+    geomdefns = (ArchGDAL.getgeomdefn(featuredefn, i) for i in 0:ngeom-1)
+    
+    name = []
+    v = []
+    for field_no in 0:nfield-1
+        field = AG.getfielddefn(featuredefn, field_no)
+        push!(name, AG.getname(field))
+    end
+    geom_names = [ArchGDAL.getname(geomdefn) for geomdefn in geomdefns]
```

Comment:
out of curiosity, can this be a generator rather than a vector?

Reply:
Yeah I wanted to to the same but https://github.com/yeesian/ArchGDAL.jl/pull/118/files/fbdef4929e9294b51ccb91c8c3a90e546a4b149c#diff-17226d5590764ff7e308bfa5045dacfaR49 part which checks for geometry names doesn't accept generators for some reason.